### PR TITLE
🐛 show real image ratio in gallery redesign

### LINKED ISSUE #4647: Mobile design on redesigned item
New redesign item mobile cuts image?

https://beta.kodadot.xyz/bsx/gallery/3579963603-32?redesign=true
image

old whole picture

image

### PR DESCRIPTION
**Pull request by petersopko**
**Thank you for your contribution** to the [KodaDot NFT gallery](https://kodadot.xyz).

👇 \_ Let's make a quick check before the contribution.

## PR Type

- [x] Bugfix
- [ ] Feature
- [ ] Refactoring

## Context

- [x] Closes #4647 
- [ ] Requires deployment <>

#### Before submitting pull request, please make sure:

- [x] My contribution builds **clean without any errors or warnings**
- [x] I've merged recent default branch -- **main** and I've no conflicts
- [x] I've tried to respect high code quality standards
- [x] I've didn't break any original functionality
- [x] I've posted a screenshot of demonstrated change in this PR

#### Optional

- [x] I've tested it at </bsx/collection>
- [x] I've tested PR on mobile
- [ ] I've written unit tests 🧪
- [ ] I've found edge cases

#### Had issue bounty label?

- [x] Fill up your KSM address: [Payout](https://beta.kodadot.xyz/transfer/?target=FNyt7T1xdbhN7dagf7yGYCRJuE3R45VmTCE3tjzy1rKxa7y)

#### Community participation

- [x] [Are you at KodaDot Discord?](https://discord.gg/35hzy2dXXh)

## Screenshot 📸

- [x] My fix has changed **something** on UI; a screenshot is best to understand changes for others.
image
image

### CODE REVIEW

**Comment by petersopko** [comment]
@exezbcz, there isn't a design for wide images in Figma; how do we want to handle it? Is this way okay?

**Comment by exezbcz** [comment]
for mobile it looks ok but on desktop it destroys the layout. I was thinking about making the img fit(width/height) to the rectangle. So we eliminate changing the layout. 
like this: 
image

Is that okay?

**Comment by petersopko** [comment]
I didn't want to add the event listener, so it won't resize responsively (you need to refresh between mobile and desktop), but that's usually only possible with devtools or if somebody wants to make their screen super small. I think it's okay, but maybe @roiLeo can take a look.

Also, I don't know if artists creating full-width pictures will be happy to have those images fitted inside a square. 

image
image

**Review by roiLeo — COMMENTED** [review]
tested on `/rmrk/gallery/13528762-f091ff4a3f410ee872-ROXYNFT_FUN_BANNERS-FUN_BANNER_030-0000000000000030?redesign=true` 

With this change many items will now have "black square" frame, I'm not sure this is the best option

ex:
`/rmrk/gallery/10371533-b6c51bd8c8868edc70-6ML5P-KUSAMA_WONDERLAND_037_OR_OCCUPIED-0000000000000043?redesign=true`
Screenshot 2023-01-10 at 10-10-03 KodaDot - Kusama NFT Market Explorer

or `/snek/gallery/185770742-2?redesign=true`

Screenshot 2023-01-10 at 10-11-22 KodaDot - Kusama NFT Market Explorer

**Comment by exezbcz** [comment]
hmm, my bad. 

I thought that like 90% are square images and different ratios are quite rare. Not sure what is the best solution for this. 
Adding bars for the full widht img is not that bad but it does not work for imgs that are portrait oriented. 

Possible solution is add a full screen button to the top right button section? Then it would be ok if we scale-crop the art to gallery img box. 

Again, what are your thoughts? Also curious about the data, how many images are not square?

**Comment by roiLeo** [comment]
> Possible solution is add a full screen button to the top right button section? Then it would be ok if we scale-crop the art to gallery img box.

In #4593 you've added it under the img box
Or we can add it in the top right like on OS

> Again, what are your thoughts? 

I would be in favour of removing the black stripes
First solution proposed by peter might be good, but we should still check if it doesn't disturb display of "square" images

> Also curious about the data, how many images are not square?

I don't think you can tell, but you have to assume that any image size can be added

File: libs/ui/src/components/MediaItem/type/ImageMedia.vue [hunk 1]
```diff
@@ -14,10 +14,11 @@ defineProps<{
   alt?: string
   original: boolean
 }>()
-</script>
 
-<style scoped>
-figure > img.image-media__image {
-  object-fit: cover;
+const isMobile = ref(window.innerWidth < 768)
+</script>
```
Comment (vikiival) [inline libs/ui/src/components/MediaItem/type/ImageMedia.vue:19]: missing new line

File: libs/ui/src/components/MediaItem/type/ImageMedia.vue [hunk 2]
```diff
@@ -14,10 +14,11 @@ defineProps<{
...
+<style>
```
Comment (vikiival) [inline libs/ui/src/components/MediaItem/type/ImageMedia.vue:20]: missing scoped or better targeting of image

**Review by vikiival — CHANGES_REQUESTED** [review]

**Comment by petersopko** [comment]
> I would be in favour of removing the black stripes
First solution proposed by peter might be good, but we should still check if it doesn't disturb display of "square" images

So are we going with this version? Can somebody confirm what should the final outcome here look like?

**Comment by exezbcz** [comment]
> > Possible solution is add a full screen button to the top right button section? Then it would be ok if we scale-crop the art to gallery img box.
> 
> In #4593 you've added it under the img box Or we can add it in the top right like on OS
>
yup, that was only quick idea and i posted it. Nothing rock solid imo :D  
> > Again, what are your thoughts?
> 
> I would be in favour of removing the black stripes First solution proposed by peter might be good, but we should still check if it doesn't disturb display of "square" images
> 
Well, its not that bad removing the stripes, not against it. What if we add the fullscreen button before production release? 
I might be wrong, but i think the entire nft experience is still tailored to squares. Possible adding "art" view in the future? Will switch your layout and make the nft fit top part without cropping.
> > Also curious about the data, how many images are not square?
> 
> I don't think you can tell, but you have to assume that any image size can be added

yup


Sooo, lets handle it like on the nft cards?

**Comment by petersopko** [comment]
> So are we going with this version? Can somebody confirm what should the final outcome here look like? 

I'm sorry if I'm misunderstanding here, but can somebody tell me what should be done here to finish up? 

I don't want to stall, but I didn't get the sense the discussion was over.

**Comment by roiLeo** [comment]
> > So are we going with this version? Can somebody confirm what should the final outcome here look like?
> 
> I'm sorry if I'm misunderstanding here, but can somebody tell me what should be done here to finish up?
> 
> I don't want to stall, but I didn't get the sense the discussion was over.

**before**
Screenshot 2023-01-13 at 16-32-57 KodaDot - Kusama NFT Market Explorer

**after**
Screenshot 2023-01-13 at 16-32-41 KodaDot - Kusama NFT Market Explorer

if it's ok for @exezbcz it's ok for me but you still got some small code changes request

**Comment by petersopko** [comment]
updated 👍🏻

File: libs/ui/src/components/MediaItem/type/ImageMedia.vue [hunk 3]
```diff
@@ -1,5 +1,7 @@
 <template>
-  <figure :class="[!original ? 'is-square image' : '']">
+  <figure
+    class="image-container"
+    :class="[!original && !isMobile ? 'is-square image' : '']">
```
Comment (roiLeo) [inline libs/ui/src/components/MediaItem/type/ImageMedia.vue:4]:
```suggestion
    :class="{ 'is-square image': !original && !isMobile }">
```

**Review by roiLeo — APPROVED** [review]
✅ code lgtm

**Review by vikiival — APPROVED** [review]

**Comment by vikiival** [comment]
pay 40 usd

**Comment by petersopko** [comment]
>  🔗 [0x92ba0920a80f39c39c1d9a655e0e7badb10adce4e06e78af9672dcb3478ab211](https://kusama.subscan.io/extrinsic/0x92ba0920a80f39c39c1d9a655e0e7badb10adce4e06e78af9672dcb3478ab211)

👀

**Comment by vikiival** [comment]
Mr @yangwao was notified

**Comment by yangwao** [comment]
me & my few notifications



yup registered, will proceed

image

**Comment by yangwao** [comment]
pay 40

**Pull request merged**